server/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/PerTestHelper.scala

```diff
@@ -124,10 +124,11 @@ private[ejea] class PerTestHelper(implicit val system: ActorSystem) extends Mock
 
   def buildEJEA(restarting: Boolean = true,
                 callCachingMode: CallCachingMode = CallCachingOff,
-                callCachingMaxFailedCopyAttempts: Int = 1000000)
+                callCachingMaxFailedCopyAttempts: Int = 1000000,
+                backendConfigurationDescriptor: BackendConfigurationDescriptor = TestConfig.emptyBackendConfigDescriptor)
```

Comment:
Maybe instead of an entire config descriptor, you could just take a `diff` from the default.

I think that would make it more obvious what is being changed in the test case, and make writing the test case easier too. Eg rather than having to do:
```
        val overriddenHogFactorAttributeString = s"hog-factor: $expectedHogFactorValue"
        val backendWithOverriddenHogFactorConfigDescriptor = BackendConfigurationDescriptor(backendConfig = ConfigFactory.parseString(overriddenHogFactorAttributeString), globalConfig)
        ejea = helper.buildEJEA(restarting = restarting, backendConfigurationDescriptor = backendWithOverriddenHogFactorConfigDescriptor)
```
Could we just say:
```
ejea = helper.buildEJEA(restarting = restarting, backendConfigurationDiff = s"hog-factor: $expectedHogFactorValue")
```

Reply:
IMHO, the potential benefit from using diff here is overweighted by the uncertainty it would introduce: e.g., what should we do if some if the configuration keys defined in diff are already defined in config - override or not?